File: 169. Majority Element/solution.go
package main

import "fmt"

func majorityElement(nums []int) int {
	seen := make(map[int]int)
	for _, num := range nums {
		value, exists := seen[num]
		if !exists {
			seen[num] = 1
		} else {
			seen[num] = value + 1
		}
		if value >= len(nums)/2 {
			return num
		}
	}
	return -1
}

func main() {
	nums := []int{3, 2, 3}
	result := majorityElement(nums)
	fmt.Println(result)
}

File: 169. Majority Element/solution.py
class Solution:
    def majorityElement(self, nums: list[int]) -> int:
        ans = 0
        count = 0
        for num in nums:
            if count == 0:
                count += 1
                ans = num
            elif ans != num:
                count -= 1
            else:
                count += 1
        return ans
